### PR TITLE
Add states for Germany

### PR DESCRIPTION
**Pull request by tyrann0us**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Add states for Germany. They are not required for shipping, but the list of states should be as complete as possible. Source: https://www.iso.org/obp/ui/#iso:code:3166:DE

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Add states for Germany

### CODE REVIEW

File: i18n/states.php [hunk 1]
```diff
@@ -304,7 +304,24 @@
...
-	'DE' => array(),
+	'DE' => array(
+		'DE-BW' => __( 'Baden-Württemberg', 'woocommerce' ),
+		'DE-BY' => __( 'Bavaria', 'woocommerce' ),
+		'DE-BE' => __( 'Berlin', 'woocommerce' ),
+		'DE-BB' => __( 'Brandenburg', 'woocommerce' ),
+		'DE-HB' => __( 'Bremen', 'woocommerce' ),
+		'DE-HH' => __( 'Hamburg', 'woocommerce' ),
+		'DE-HE' => __( 'Hesse', 'woocommerce' ),
+		'DE-MV' => __( 'Mecklenburg-Western Pomerania', 'woocommerce' ),
```
Comment (barryhughes) [inline i18n/states.php]: Noting here that we conform to Unicode CLDR, it seems this should be *Mecklenburg-Vorpommern* [(see here)](https://github.com/unicode-org/cldr/blob/e717766cefb06411b35e4b63953a2b29686d66d2/common/subdivisions/en.xml#L1098):

```suggestion
		'DE-MV' => __( 'Mecklenburg-Vorpommern', 'woocommerce' ),
```

**Comment by barryhughes** [comment]
Thanks for this change! I noted one minor issue and it is probably also worth rebasing this on `trunk` (to resolve the conflict in `i18n/states.php`), though we can help with that if you experience any issues.

**Comment by tyrann0us** [comment]
@barryhughes, thanks for checking the PR! It's the first time I need to rebase from upstream and I totally failed to do so. Sorry for the mess. If possible, can you clean this up and do a clean rebase? Otherwise, I'd just close this PR and open a new one … 😉

**Comment by barryhughes** [comment]
Sure thing!

**Comment by barryhughes** [comment]
@tyrann0us — apologies but, actually, I think a clean PR will be the fastest/cleanest/safest way to address this one.

**Comment by tyrann0us** [comment]
Superseded by #31825.

**Pull request closed without merging**